crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -464,243 +511,235 @@ end
-#!
-#! Inputs:  [index, KEY, NEW_VALUE]
+#! Inputs:  [name_id_prefix, name_id_suffix, KEY, NEW_VALUE]
 #! Outputs: [OLD_MAP_ROOT, OLD_MAP_VALUE]
 #!
 #! Where:
-#! - index is the index of the storage slot which contains the map root.
+#! - name_id_{prefix, suffix} are the prefix and suffix felts of the name identifier, which are
+#!   the first two felts of the hashed slot name.
+#!    - the slot must point to the root of the storage map.
 #! - NEW_VALUE is the value to set under KEY.
 #! - KEY is the key to set.
 #! - OLD_MAP_VALUE is the previous value of the item.
 #! - OLD_MAP_ROOT is the root of the old map before insertion
 #!
 #! Panics if:
+#! - a slot with the provided name ID does not exist in account storage.
 #! - the storage slot type is not map.
-#! - no map is found for ROOT.
-export.set_map_item.12
-    # store index for later
-    dup loc_store.0 exec.get_item
-    # => [OLD_ROOT, KEY, NEW_VALUE, ...]
-
-    movdnw.2 loc_load.0
-    # => [index, KEY, NEW_VALUE, OLD_ROOT, ...]
+#! - no map with the root of the slot is found.
+export.set_map_item
+    exec.memory::get_account_storage_slots_section_ptr
+    # => [storage_slots_ptr, name_id_prefix, name_id_suffix, KEY, NEW_VALUE]
 
-    emit.ACCOUNT_STORAGE_BEFORE_SET_MAP_ITEM_EVENT
-    # => [index, KEY, NEW_VALUE, OLD_ROOT, ...]
+    # resolve the slot name to its pointer
+    exec.find_storage_slot
+    # => [slot_ptr, KEY, NEW_VALUE]
 
-    # check if storage type is map
-    exec.get_storage_slot_type_by_index
-    # => [slot_type, KEY, NEW_VALUE, OLD_ROOT]
+    # load the slot type
+    dup add.ACCOUNT_SLOT_TYPE_OFFSET mem_load
+    # => [slot_type, slot_ptr, KEY, NEW_VALUE]
 
-    # check if slot_type == map
+    # assert slot_type is map
     exec.constants::get_storage_slot_type_map eq
     assert.err=ERR_ACCOUNT_SETTING_MAP_ITEM_ON_NON_MAP_SLOT
-    # => [KEY, NEW_VALUE, OLD_ROOT]
+    # => [slot_ptr, KEY, NEW_VALUE]
 
-    # duplicate the original KEY and the NEW_VALUE to be able to emit an event after the
-    # account storage item was updated
-    movupw.2 dupw.2 dupw.2
-    # => [KEY, NEW_VALUE, OLD_ROOT, KEY, NEW_VALUE]
+    exec.set_map_item_raw
+    # => [OLD_MAP_ROOT, OLD_MAP_VALUE]
+end
+
+#! Stores NEW_VALUE under the specified KEY within the map contained in the given account storage slot.
+#!
+#! WARNING: The slot ptr must point to a map slot.
+#!
+#! Inputs:  [slot_ptr, KEY, NEW_VALUE]
+#! Outputs: [OLD_MAP_ROOT, OLD_MAP_VALUE]
+#!
+#! Where:
+#! - slot_ptr is the pointer to a slot.
+#! - KEY is the key to set.
+#! - NEW_VALUE is the value to set under KEY.
+#! - OLD_MAP_VALUE is the previous value of the item.
+#! - OLD_MAP_ROOT is the root of the old map before insertion
+#!
+#! Panics if:
+#! - a slot with the provided name ID does not exist in account storage.
+#! - no map with the root of the slot is found.
+#!
+#! Locals:
+#! - 0: slot_ptr
+#! - 4..8: OLD_MAP_VALUE
+#! - 8..12: OLD_MAP_ROOT
+export.set_map_item_raw.12
```

Comment:
Should this not be just `proc`? It doesn't seem like we use it anywhere outside of this module.